src/components/layout/genericDialog/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
+
+interface IProps {
+    open: boolean,
+    title: string,
+    description: string,
+    fields: Array<any>,
+    handleChange: any,
+    handleSubmit: any,
+    handleClose: any
+}
+
+function GenericDialog({ open, title, description, fields = [], handleChange, handleSubmit, handleClose }: IProps): JSX.Element {
+    return (
+        <Dialog open={open} onClose={() => "handleClose"} aria-labelledby="form-dialog-title">
+            <DialogTitle id="form-dialog-title">{title}</DialogTitle>
+            <DialogContent>
+                <DialogContentText>
+                    {description}
+                </DialogContentText>
+                {fields.map((field: any, index: number) => (
+                    <TextField
+                        key={index}
```

Comment:
It is not a good practice to use index as a key. Maybe `field.name` is a good key
```suggestion
                        key={field.name}
```

Reply:
https://medium.com/wesionary-team/using-index-as-a-key-is-an-anti-pattern-in-react-8e5db3aea3a9

Reply:
Agree